Pod log output is not passed on time to caller

We are having a strange issue with latest kubedock in gradle build:
1. testcontainers is creating a postgres pod for test execution, it is initialized ok, pog log shows that database is waiting for connections [kubedock-pod.log](https://github.com/user-attachments/files/18183263/kubedock-pod.log)
2. but test is still waiting for kubedock to return that log output and confirm that pod is ready, that takes 1 minute sharp 
[gradle-buildah-agent.log](https://github.com/user-attachments/files/18183355/gradle-buildah-agent.log)
3. after that timeout test is considered failed:
```
Caused by: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image postgres:16.3
	at app//org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:359)
	at app//org.testcontainers.containers.GenericContainer.start(GenericContainer.java:330)
	at app//org.testcontainers.jdbc.ContainerDatabaseDriver.connect(ContainerDatabaseDriver.java:134)
	at app//com.zaxxer.hikari.util.DriverDataSource.getConnection(DriverDataSource.java:137)
	at app//com.zaxxer.hikari.pool.PoolBase.newConnection(PoolBase.java:360)
	at app//com.zaxxer.hikari.pool.PoolBase.newPoolEntry(PoolBase.java:202)
	at app//com.zaxxer.hikari.pool.HikariPool.createPoolEntry(HikariPool.java:461)
	at app//com.zaxxer.hikari.pool.HikariPool.checkFailFast(HikariPool.java:550)
	... 60 more
Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception
	at app//org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:88)
	at app//org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:344)
	... 67 more
Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container
	at app//org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:563)
	at app//org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:354)
	at app//org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
	... 68 more
Caused by: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for log output matching '.*database system is ready to accept connections.*\s'
	at app//org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy.waitUntilReady(LogMessageWaitStrategy.java:47)
	at app//org.testcontainers.containers.wait.strategy.AbstractWaitStrategy.waitUntilReady(AbstractWaitStrategy.java:52)
	at app//org.testcontainers.containers.PostgreSQLContainer.waitUntilContainerStarted(PostgreSQLContainer.java:149)
	at app//org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:500)
```

--- a/internal/util/ioproxy/ioproxy.go
+++ b/internal/util/ioproxy/ioproxy.go
@@ -5,6 +5,7 @@ import (
 	"io"
 	"k8s.io/klog"
 	"sync"
+	"time"
 )
 
 // StdType is the type of standard stream
@@ -49,8 +50,13 @@ func (w *IoProxy) Write(p []byte) (int, error) {
 	w.buf = append(w.buf, p...)
 	for w.process() != 0 {
 	}
-	// callers of ipProxy.New() should call Flush() before the request returns.
-	// This can be guaranteed by doing a defer Flush() directly after ioProxy.New()
+	if len(w.buf) > 0 && !w.flusher {
+		w.flusher = true
+		go func() {
+			time.Sleep(100 * time.Millisecond)
+			w.Flush()
+		}()
+	}
 	return len(p), nil
 }
 
@@ -99,6 +105,9 @@ func (w *IoProxy) write(p []byte) error {
 func (w *IoProxy) Flush() error {
 	w.lock.Lock()
 	defer w.lock.Unlock()
+	if len(w.buf) == 0 {
+		return nil
+	}
 	err := w.write(w.buf)
 	w.buf = []byte{}
 	w.flusher = false

